### ISSUE TITLE
setupAutomaticPushTracking() causes app to stop receiving push notifications when not running

### ISSUE THREAD

**Issue by mluisbrown** [mixpanel/mixpanel-swift#201]
The `setupAutomaticPushTracking()` [method](https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel/AutomaticEvents.swift#L162) prevents an app from receiving Push Notifications when the app is not running.

It is called [here](https://github.com/mixpanel/mixpanel-swift/blob/master/Mixpanel/AutomaticEvents.swift#L87) as part of initialisation. I couldn't find the exact cause, but something about the swizzling is breaking APNs when the app is not running. 

We have had to create a fork of this repo removing this call in order for our app to continue working.

**Comment by zihejia** [comment]
Thanks @mluisbrown for raising this, we will have a look.

**Comment by daniloaliberti** [comment]
Hi @zihejia 👋, 
Is there any good news on this please?

**Comment by zihejia** [comment]
For AutomaticPushTracking, we use swizzling to add tracking to some delegate methods for receiving notification. I suspect If you have some other 3rd party libraries swizzle the same method like ' ..didReceiveRemoteNotification...', '...didReceiveNotificationResponse...'. It will cause conflicts and crash the app silently in the background so the app won't receive Push Notifications.  Is that the case for you @daniloaliberti or @mluisbrown ?

**Comment by zihejia** [comment]
Add a related issue from the past: https://github.com/mixpanel/mixpanel-swift/issues/119 for reference

**Comment by mluisbrown** [comment]
@zihejia niether our code not any 3rd party libraries we use (other than Mixpanel) are doing any swizzling.

**Comment by zihejia** [comment]
hi @mluisbrown, do you mind to share the implementation of your notification delegate method or what you do in didReceiveRemoteNotification delegate? The swizzle method will call your origin method and add tracking on top of it. Not sure if there is something will cause the problem.

**Comment by mluisbrown** [comment]
@zihejia as I mentioned in [the other issue](https://github.com/mixpanel/mixpanel-swift/issues/119#issuecomment-374569527):

With the current Mixpanel code, in the scenario of "notification received when app not running", neither the swizzled Mixpanel method:

```swift
@objc func userNotificationCenter(_ center: UNUserNotificationCenter,
                                  newDidReceive response: UNNotificationResponse,
                                  withCompletionHandler completionHandler: @escaping () -> Void)
```

nor the original delegate implementation is called.

**Comment by mikkelam** [comment]
I use firebase cloud messaging that swizzles and i had the crash

**Comment by zihejia** [comment]
We can provide a new init method with automatic push tracking disabled to avoid this situation.

**Comment by mluisbrown** [comment]
@zihejia we don't use any 3rd party library for push notifications. 

> We can provide a new init method with automatic push tracking disabled to avoid this situation.

That would be awesome 👍

**Comment by zihejia** [comment]
UPDATE: We introduced the new init method since v2.4.2. which can turn `automaticPushTracking` off.
`init(apiToken: String?, launchOptions: [UIApplicationLaunchOptionsKey : Any]?, flushInterval: Double, name: String, automaticPushTracking: Bool = true, optOutTrackingByDefault: Bool = false)`

We have just released v2.5.0 which supports Xcode 10 and swift 4.2, might worth to have a try for this problem if it's still relevant to you.
